influxdb3_catalog/src/catalog.rs

```diff
@@ -1139,6 +1133,13 @@ impl TableDefinition {
             .get_by_right(&name.into())
             .and_then(|id| self.columns.get(id).map(|def| (*id, def)))
     }
+
+    pub fn series_key(&self) -> Vec<String> {
+        self.series_key
+            .iter()
+            .map(|k| self.column_id_to_name_unchecked(k).to_string())
+            .collect()
+    }
```

Comment:
Since this gets called in the write path for each line, might be worth returning a `Vec<&str>` or `Vec<Arc<str>>` to avoid the string copies. Or even a slice, if possible.

Furthermore, having to do the lookup by ID for the name every time could also be avoided by holding the `Arc<str>` names around in the `TableDefinition` then just iterating over those directly.

Depending on how far you take it, this could lead to a substantial change if you had to change the `TableDefinition` struct, so might be better for a follow-on if that's the case.

Reply:
Just from a cursory try I do think it'll be a bit more substantial, especially if we cache those keys in the `TableDefinition`. I've opened up an issue for future work https://github.com/influxdata/influxdb/issues/25606

Reply:
yeah, returning the series key should be as cheap as possible. Given that every table now has one, it should just be a reference that gets returned (or an Arc'd thing?).